engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/resolve/TypeFunctions.scala

```diff
@@ -106,16 +106,14 @@ case object TypeFunctions extends IRPass {
     * @return `expr`, with any typing functions resolved
     */
   def resolveExpression(expr: Expression): Expression = {
-    expr.transformExpressions {
-      case asc: Type.Ascription => asc
```

Comment:
This line was causing `transformExpressions` to _not_ go deeper into the `Type.Ascription`.

I'm not sure why it was there. It was breaking the processing of transforming the expression `a | b` or `a ! b` from `Operator.Binary` into `Application.Prefix` and later into `type.Set.Union` or `Type.Error` which meant later passes were not detecting these constructs correctly.

Reply:
I hope changing that does not break any other tests 😅

Reply:
Actually it **did break** one test.

My changes to make this test pass again are in commit aeff369af7818ca673b75f1f62d916b3cc630148

@JaroslavTulach would you mind to take a look? As I think you wrote this test originally.
What changed is that the `arg2` is no longer an `Application.Prefix` but now the `Literal`, _but_ it does have the correct type ascription - so from my perspective this behaviour is still valid, just slightly different. But please double check if I did not misinterpret the true intention behind this test.